data_pipeline/utils/data_store/bq_data_service.py

```diff
@@ -349,3 +351,70 @@ def load_new_data_in_temp_table_to_actual_table(
     query_job.result()  # Wait for the job to complete.
 
     LOGGER.info("Loaded table %s", table_id)
+
+
+def get_distinct_values_from_bq(
+            project_name: str,
+            dataset_name: str,
+            column_name: str,
+            table_name_source: str,
+            table_name_for_exclusion: str = None
+        ) -> pd.DataFrame:
+
+    sql = """
+        SELECT DISTINCT {{ column_name }}
+        FROM  `{{ project_name }}.{{ dataset_name }}.{{ table_name_source }}`
+        {% if table_name_for_exclusion %}
+        WHERE {{ column_name }} NOT IN
+            (
+                SELECT {{ column_name }}
+                FROM `{{ project_name }}.{{ dataset_name }}.{{ table_name_for_exclusion }}`
+            )
+        {% endif %}
+    """
+    params = {
+        'project_name': project_name,
+        'dataset_name': dataset_name,
+        'column_name': column_name,
+        'table_name_source': table_name_source,
+        'table_name_for_exclusion': table_name_for_exclusion
+        }
+
+    jin = JinjaSql(param_style='pyformat')
+    query, bind_params = jin.prepare_query(sql, params)
+    client = get_bq_client(project=project_name)
+    query_job = client.query(query % bind_params)  # Make an API request.
+    results = query_job.result()  # Waits for query to finish
+
+    return pd.concat([pd.DataFrame([row.column]) for row in results], ignore_index=True)
```

Comment:
minor: now that this is "shared", would it now be worth to just very slightly improve this line?
personally I would just return a list of values, i.e.

```python
return [row.column for row in results]
```

If you want to use Pandas, then Series seems to be the more natural choice as it's just one list of values, e.g.:

```python
return pd.Series([row.column for row in results])
```

Or with a DataFrame (but it wouldn't be obvious what the column name in the result is):

```python
return pd.Series([row.column for row in results]).to_frame(column_name)
```